src/GraphUtils.ts

```diff
@@ -510,76 +485,129 @@ export function translateCurve(curve: Curve, dx: number, dy: number, canvasPrope
+        withinXBoundary = (rightImportant[0] - mousePosition[0]) > XBUFFER && (mousePosition[0] - leftImportant[0]) > XBUFFER;
+        withinYBoundary = (isMaxima && ((rightImportant[1] - mousePosition[1]) > YBUFFER && (leftImportant[1] - mousePosition[1]) > YBUFFER)) || (!isMaxima && ((mousePosition[1] - rightImportant[1]) > YBUFFER && (mousePosition[1] - leftImportant[1]) > YBUFFER));
+    } else {
+        const definedImportant = prevImportant ?? nextImportant;
+        if (!isDefined(definedImportant)) return;
+        if (definedImportant[0] - currImportant[0] > 0) {
+            withinXBoundary = (definedImportant[0] - mousePosition[0]) > XBUFFER;
+        } else {
+            withinXBoundary = (mousePosition[0] - definedImportant[0]) > XBUFFER;
+        }
+        if (definedImportant[1] - currImportant[1] > 0) {
+            withinYBoundary = (definedImportant[1] - mousePosition[1]) > YBUFFER;
+        } else {
+            withinYBoundary = (mousePosition[1] - definedImportant[1]) > YBUFFER;
+        }
+    }
+
+    if (isDefined(currImportant) && (withinXBoundary || withinYBoundary)) {
         // to this point we get the clicked knot and the turning/end points either side, now we will split the curve into the two
         // origional max/min sides and the 2 new curves to be stretched, then combine them all after.
         let leftStaticPoints = new Array<Point>();
-        let rightStaticPoints = new Array<Point>();
         let leftStretchedCurve = new Curve;
         let rightStretchedCurve = new Curve;
-        for (let t = selectedCurve.pts.length-1; t > -1; t--) {
-            if (selectedCurve.pts[t][0] > tempMax[0]) {
-                rightStaticPoints.push(selectedCurve.pts[t]);
-                selectedCurve.pts.pop(/* selectedCurve.pts[t] */);
-            } else if (selectedCurve.pts[t][0] <= tempMax[0] && selectedCurve.pts[t][0] >= turningPoints[selectedPointIndex][0]) {
-                rightStretchedCurve.pts.push(selectedCurve.pts[t]);
-                selectedCurve.pts.pop(/* selectedCurve.pts[t] */);
-            } else if (selectedCurve.pts[t][0] <= turningPoints[selectedPointIndex][0] && selectedCurve.pts[t][0] >= tempMin[0]) {
-                leftStretchedCurve.pts.push(selectedCurve.pts[t]);
-                selectedCurve.pts.pop(/* selectedCurve.pts[t] */);
-            } else if (selectedCurve.pts[t][0] < tempMin[0]) {
-                leftStaticPoints.push(selectedCurve.pts[t]);
-                selectedCurve.pts.pop(/* selectedCurve.pts[t] */);
-            } else {
-                selectedCurve.pts.pop(/* selectedCurve.pts[t] */);
+        let rightStaticPoints = new Array<Point>();
+
+        if (!withinXBoundary) {
+            mousePosition[0] = currImportant[0];
+        } else if (!withinYBoundary) {
+            mousePosition[1] = currImportant[1];
+        }
+
+        let currentPointList = (prevImportant) ? 0 : 1;
```

Comment:
```suggestion
        let currentSectionState= (prevImportant) ? 0 : 1;
```
And refactor the usages for easier understanding